File: lib/data/model/downloaded_image.dart
import 'package:meta/meta.dart';

///
/// Contains info about a successfully downloaded image.
///
class DownloadedImage {
  /// Image url.
  final String url;

  /// Path to the downloaded image file.
  final String path;

  /// Hash calculated from the image.
  /// Can be an empty String (for the download history).
  final String hash;

  DownloadedImage({
    @required this.url,
    @required this.path,
    @required this.hash,
  });

  DownloadedImage.fromJson(Map<String, dynamic> json)
      : this.url = json['url'],
        this.path = json['path'],
        this.hash = json['hash'];
}

File: android/app/src/main/kotlin/com/example/image_hasher/data/model/DownloadedImage.kt
package com.example.image_hasher.data.model

import android.util.Base64
import org.json.JSONObject
import java.io.File

/**
 * Contains info about a successfully downloaded image.
 */
data class DownloadedImage(
       val url: String,
       val file: File,
       val hash: String = ""
) {

   fun toJson() = JSONObject(mapOf(
          "url" to url,
          "path" to file.path,
          "hash" to hash
   ))

   fun calculateBase64(): DownloadedImage {
      val base64 = Base64.encodeToString(file.readBytes(), Base64.DEFAULT)
      return DownloadedImage(url, file, base64)
   }
}
